### PR TITLE
ocspapitest.c: check the return of three memory-allocation related functions

### PR DESCRIPTION
**Pull request by x2018**
Check for the return X509_NAME_new(), ASN1_BIT_STRING_new() and ASN1_INTEGER_new(), respectively, because they return NULL when fail.
Especially for ASN1_BIT_STRING_new() and ASN1_INTEGER_new(), the memory may be wrongly access by ASN1_BIT_STRING_set() and ASN1_INTEGER_set_uint64() since they do not check the validity of the first argument.

### CODE REVIEW

**Review by t8m — APPROVED** [review]

**Review by kroeckx — APPROVED** [review]

File: test/ocspapitest.c [hunk 1]
```diff
@@ -78,7 +78,10 @@ static OCSP_BASICRESP *make_dummy_resp(void)
     ASN1_BIT_STRING *key = ASN1_BIT_STRING_new();
     ASN1_INTEGER *serial = ASN1_INTEGER_new();
 
-    if (!X509_NAME_add_entry_by_NID(name, NID_commonName, MBSTRING_ASC,
+    if (!TEST_ptr(name)
+        || !TEST_ptr(key)
+        || !TEST_ptr(serial)
+        || !X509_NAME_add_entry_by_NID(name, NID_commonName, MBSTRING_ASC,
```
Comment (paulidale) [inline test/ocspapitest.c:84]: Any chance converting this and the two following lines to use TEST_true so the exact failure is reported?

Comment (x2018) [reply test/ocspapitest.c]: Changed.

**Review by paulidale — APPROVED** [review]
Thanks

**Review by t8m — APPROVED** [review]
@kroeckx still OK?

**Comment by t8m** [comment]
Merged to master and 3.0 branches after rewording the commit messages slightly.

**Pull request closed without merging**